src/e3/anod/qualifiers_manager.py

```diff
@@ -594,6 +639,21 @@ def compute_build_space_name(self) -> None:
             hash_obj.update("_".join(hash_pool).encode("utf-8"))
             bs.append(hash_obj.hexdigest()[:8])
 
+        if self.add_target_info_to_bs and self.anod_instance.env.is_cross:
+            # We can run test on many different cross OS version
+            # from the same host machine, make sure that we have a
+            # different build space each time
+            os_version = self.anod_instance.env.target.os.version
+            if os_version and os_version != "unknown":
+                bs.append(self.os_version_aliases.get(os_version, os_version))
+            # We also can test on different machines, so add it as well
+            machine = self.anod_instance.env.target.machine
+            if machine and machine != "unknown":
+                # and make sure the machine name is compatible with
+                # file paths: only keep letters
+                machine_str = "".join([c for c in machine if c.isalpha()])
```

Comment:
Why not `isalnum()` ?

Reply:
For me, it's neither isalnum nor isalpha.

A machine can be any string, it can have dash inside, for example.

Reply:
I think the logic was taken from existing code

Reply:
Maybe we can indeed accept more characters

Reply:
Will adjust the change to accept numbers and -

Reply:
Solved